internal/services/verifible_credential_service.go

```diff
@@ -0,0 +1,194 @@
+package services
+
+//go:generate mockgen -source verifible_credential_service.go -destination mocks/verifible_credential_service_mock.go
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	ethC "github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/crypto/secp256k1"
+	"github.com/mr-tron/base58"
+	"github.com/piprate/json-gold/ld"
+	"github.com/volatiletech/null/v8"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+
+	"github.com/DIMO-Network/devices-api/internal/config"
+	"github.com/DIMO-Network/devices-api/models"
+	"github.com/DIMO-Network/shared/db"
+	"github.com/twinj/uuid"
+)
+
+type VerifiableCredentialService struct {
+	dbs                db.Store
+	Issuer             string
+	PublicKey          ecdsa.PublicKey
+	PrivateKey         ecdsa.PrivateKey
+	KeyEnc             string
+	VerificationMethod string
+	NFTAddress         common.Address
+	ChainID            *big.Int
+}
+
+var secp256k1Prefix = []byte{0xe7, 0x01}
+var period byte = '.'
+
+type VCService interface {
+	CreateVinCredential(vin string, tokenID *big.Int) (string, error)
+}
+
+// NewVerifiableCredentialService generates vc service
+func NewVerifiableCredentialService(dbs db.Store, settings *config.Settings) (VerifiableCredentialService, error) {
+	xb, err := base64.RawURLEncoding.DecodeString(settings.DIMOKeyX)
```

Comment:
We will probably use a KMS eventually. For now maybe base64-encode a private key, decode that, run

```
priv, _ := crypto.ToECDSA(b)
```

Compressed public key is then `crypto.CompressPubkey(priv.PublicKey)`.